app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# TODO: Add desc
+# desc here
+class ArticlesController < ApplicationController
+  before_action :logged_in_user, only: %i[create destroy edit update new]
+  before_action :correct_user,   only: %i[destroy edit update]
+  before_action :allowed_user,   only: %i[destroy destroy edit update]
+
+  def index
+    @article = current_user.articles.build if logged_in?
+    @articles = Article.all.paginate(page: params[:page])
+  end
+
+  def new
+    @article = Article.new
+  end
+
+  def create
+    @article = current_user.articles.build(article_params)
+    if @article.save
+      flash[:success] = 'Article created!'
+      redirect_to root_url
+    else
+      flash[:danger] = 'Article failed!'
```

Comment:
Maybe change it to a better message. It's not the article that failed, is the creation of the article that failed. So something like 'Couldn't create an article' or 'Failed when creating article'

Reply:
Fixed